index.template.html

```diff
@@ -244,31 +244,60 @@
 </style>
 
 <script>
-    document.addEventListener('DOMContentLoaded', () => {
-        // let ReSpec do syntax highlighting first
-        document.respec.ready.then(() => {
-            document.querySelectorAll('.with-default').forEach(n => {
-                let wo = n.querySelector('pre:nth-of-type(1)');
-                let w = n.querySelector('pre:nth-of-type(2)');
-                let cbox = n.querySelector('.with-default-control input[type="checkbox"]');
-
-                w.classList.add('selected');
-                cbox.checked = true;
-
-                cbox.addEventListener('change', e => {
-                    if (e.target.checked) {
-                        // with default
-                        w.classList.add('selected');
-                        wo.classList.remove('selected');
-                    } else {
-                        // without
-                        w.classList.remove('selected');
-                        wo.classList.add('selected');
-                    }
-                });
-            });
-        });
-    });
+	if (document.respec) {
+		// ReSpec mode
+		document.addEventListener('DOMContentLoaded', () => {
+			// let ReSpec do syntax highlighting first
+			document.respec.ready.then(() => {
+				document.querySelectorAll('.with-default').forEach(n => {
+					let wo = n.querySelector('pre:nth-of-type(1)');
+					let w = n.querySelector('pre:nth-of-type(2)');
+					let cbox = n.querySelector('.with-default-control input[type="checkbox"]');
+
+					w.classList.add('selected');
+					cbox.checked = true;
+
+					cbox.addEventListener('change', e => {
+						if (e.target.checked) {
+							// with default
+							w.classList.add('selected');
+							wo.classList.remove('selected');
+						} else {
+							// without
+							w.classList.remove('selected');
+							wo.classList.add('selected');
+						}
+					});
+				});
+			});
+		});
+	} else {
+		// W3C specs on the W3C server don't use ReSpec
+		document.addEventListener('readystatechange', () => {
+			if (document.readyState === 'complete') {
+				document.querySelectorAll('.with-default').forEach(n => {
+					let wo = n.querySelector('pre:nth-of-type(1)');
+					let w = n.querySelector('pre:nth-of-type(2)');
+					let cbox = n.querySelector('.with-default-control input[type="checkbox"]');
+
+					w.classList.add('selected');
+					cbox.checked = true;
+
+					cbox.addEventListener('change', e => {
+						if (e.target.checked) {
+							// with default
+							w.classList.add('selected');
+							wo.classList.remove('selected');
+						} else {
+							// without
+							w.classList.remove('selected');
+							wo.classList.add('selected');
+						}
+					});
+				});
```

Comment:
This code could probably be moved to its own function to avoid redundancy?

Reply:
good point. updated PR accordingly.